gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -330,8 +330,9 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
 
     // Check quota limits against run immediately flows or adhoc flows before saving the schedule
     // In warm standby mode, this quota check will happen on restli API layer when we accept the flow
```

Comment:
afaik `[FlowConfigV2ResourceLocalHandler](https://github.com/apache/gobblin/blob/69f3f9c33c9679d723c447fbe626db8f1b7afa9e/gobblin-service/src/main/java/org/apache/gobblin/service/modules/restli/GobblinServiceFlowConfigV2ResourceHandlerWithWarmStandby.java#L106)`  and `[FlowCatalog](https://github.com/apache/gobblin/blob/69f3f9c33c9679d723c447fbe626db8f1b7afa9e/gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java#L403 )` are not doing quota check on adding the spec. Where do we do the check in API layer?

Reply:
In warmStandyby mode, scheduler is not the listener when we add flow spec to flowCatalog. But compiler is the listener as defined [here](https://github.com/apache/gobblin/blob/master/gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java#L413). And we check quota [here](https://github.com/apache/gobblin/blob/master/gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/BaseFlowToJobSpecCompiler.java#L205)